### PR TITLE
373. Jetpack Compose "Problem filters".

### PR DESCRIPTION
**Pull request by derevyanko**

### CODE REVIEW

File: app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivityNew.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            }
+        }
+    }
+
+    @Composable
+    private fun ProblemsFilters() {
+        val swipeRefreshState = rememberSwipeRefreshState(false)
+
+        Box {
+            Box(
```
Comment (yev-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivityNew.kt]: Why do you need two boxes here?

File: app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivity.kt [hunk 2]
```diff
@@ -1,37 +1,71 @@
...
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
-        supportActionBar?.setDisplayShowHomeEnabled(true)
 
-        recyclerView.adapter = epoxyController.adapter
+        setContent {
+            AlgoismeTheme {
+                ProblemsFilters(
+                    filtersState = filtersState,
+                    onFilter = { title, isChecked ->
+                        onFilter(title = title, isChecked = isChecked)
```
Comment (yev-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivity.kt:43]: Argument labels can be omitted here as they are the same as argument names.

File: app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivity.kt [hunk 3]
```diff
@@ -44,18 +78,76 @@ class ProblemsFiltersActivity : AppCompatActivity(), StoreSubscriber<ProblemsSta
...
+    private fun onFilter(title: String, isChecked: Boolean) {
+        store.dispatch(ProblemsRequests.ChangeTagCheckStatus(title, isChecked))
+    }
+
+    @Composable
+    private fun ProblemsFilters(
+        filtersState: State<List<FilterItem>>,
+        onFilter: (String, Boolean) -> Unit
+    ) {
+        val swipeRefreshState = rememberSwipeRefreshState(false)
```
Comment (yev-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFiltersActivity.kt:90]: Can be used directly.

**Review by yev-kanivets — CHANGES_REQUESTED** [review]
Good JC overall 👍 But there a few comments to resolve before going forward.

**Comment by derevyanko** [comment]
I won't merge it with Problems task until you review it to simplify this process for you.

**Review by yev-kanivets — APPROVED** [review]

**Comment by yev-kanivets** [comment]
@derevyanko go ahead with merging! I'll review all the changes one more time.

**Pull request merged**